src/main/java/kitchenpos/menus/tobe/domain/Menu.kt

```diff
@@ -0,0 +1,67 @@
+package kitchenpos.menus.tobe.domain
+
+import jakarta.persistence.*
+import kitchenpos.common.Price
+import kitchenpos.products.domain.ProductPriceChanged
+import java.util.*
+
+@Table(name = "menu")
+@Entity
+class Menu(
+    menuGroup: MenuGroup,
+    name: String,
+    price: Price,
+    displayStatus: Boolean,
+    menuProducts: MenuProducts,
+    menuNameValidatorService: MenuNameValidatorService,
+    menuPriceValidatorService: MenuPriceValidatorService
+) {
+    @Column(name = "id", columnDefinition = "binary(16)")
+    @Id
+    val id: UUID = UUID.randomUUID()
+
+    @ManyToOne(optional = false)
+    @JoinColumn(
+        name = "menu_group_id",
+        columnDefinition = "binary(16)",
+        foreignKey = ForeignKey(name = "fk_menu_to_menu_group")
+    )
+    val menuGroup: MenuGroup = menuGroup
+
+    @Column(name = "name", nullable = false)
+    val name: String = name
+
+    @Column(name = "price", nullable = false)
+    var price: Price = price
+        private set
+
+    @Column(name = "displayed", nullable = false)
+    var displayStatus: Boolean = displayStatus
+        private set
+
+    @OneToMany(cascade = [CascadeType.PERSIST, CascadeType.MERGE], mappedBy = "menu", fetch = FetchType.LAZY)
+    val menuProducts: List<MenuProduct> = menuProducts.items
```

Comment:
MenuProducts를 `value class`로 두신다면 아래와 같이 변경할 수 있을 것 같네요. 🙂

```suggestion
    @OneToMany(cascade = [CascadeType.PERSIST, CascadeType.MERGE], mappedBy = "menu", fetch = FetchType.LAZY)
    val menuProducts: MenuProducts = MenuProducts(menuProducts.items)
```


`MenuProduct`를 Child Entity로 보셨군요?

Reply:
네넵 ㅎㅎ 
뭔가 나중에 메뉴상품의 개수를 변경시키거나 하는 요구사항이 올 수도 있을것같아서 VO 보다는 Entity로 구분지었어요

Reply:
처음에 말씀주신것처럼 해봤었는데 jpa에서 컨테이너 타입만 인식한다고 해서 적용이 힘들것같습니다 ㅜ

<img width="957" alt="Screenshot 2024-06-01 at 8 31 09 PM" src="https://github.com/next-step/ddd-tactical-design/assets/124428341/2be0220b-c021-402e-8f7d-f2d119d1f8d3">

Reply:
단순 원시값 포장 케이스는 value class도 활용이되는데, Collection은 이슈가 있나보군요 🤔

그렇다면 조금 아쉽지만 MenuProducts를 data class로 변경하고 `@Embedded` 애너테이션을 사용하는 방식으로 변경해주시면 좋을 것 같아요. (컬럼의 속성 메타 데이터들은 MenuProducts의 인스턴스 필드에 명시해주시면 될 것 같아요. 🙂)

<img width="1615" alt="image" src="https://github.com/next-step/ddd-tactical-design/assets/47850258/c5fa4e57-dc31-48e1-b5ea-2c040327c1f0">

Reply:
단순 원시값 포장 케이스는 value class도 활용이되는데, Collection은 이슈가 있나보군요 🤔

그렇다면 조금 아쉽지만 MenuProducts를 data class로 변경하고 `@Embedded` 애너테이션을 사용하는 방식으로 변경해주시면 좋을 것 같아요. (컬럼의 속성 메타 데이터들은 MenuProducts의 인스턴스 필드에 명시해주시면 될 것 같아요. 🙂)

<img width="1615" alt="image" src="https://github.com/next-step/ddd-tactical-design/assets/47850258/c5fa4e57-dc31-48e1-b5ea-2c040327c1f0">